src/bridge/MatrixHandler.ts

```diff
@@ -1066,14 +1069,43 @@ export class MatrixHandler {
                 cacheBody = reply.reply;
             }
         }
+
+        // special handling for edits
+        if (event.content["m.relates_to"] && event.content["m.relates_to"].rel_type === "m.replace") {
+            const originalEventId = event.content["m.relates_to"].event_id;
+            let originalBody = this.getCachedEvent(originalEventId)?.body;
+            if (!originalBody) {
+                try {
+                    // FIXME: this will return the new event rather than the original one
+                    // to actually see the original content we'd need to use whatever
+                    // https://github.com/matrix-org/matrix-doc/pull/2675 stabilizes on
+                    const eventContent = await this.ircBridge.getAppServiceBridge().getIntent().getEvent(
```

Comment:
This might fail for DMs, as the bot can't read events in rooms it's not part of. You might have to determine the other user in a DM room and get an intent for them.

Reply:
Good call. Is there any reason we shouldn't just always use the user's intent here?

Reply:
Fixed, and answered my own question in 021d29aa :)